File: practice/algorithms/strings/alternating_characters/AlternatingCharacters.scala
import scala.io.Source

object AlternatingCharacters extends App {
  val strings = Source.stdin.getLines().drop(1)

  println(strings.map(deletions).mkString("\n"))

  def deletions(s: String):Int = {
    var last = s.charAt(0)
    var dels = 0
    s.drop(1).foreach {
      c => {
        if (last == c) {
          dels += 1
        } else {
          last = c
        }
      }
    }
    dels
  }
}

File: practice/algorithms/strings/alternating_characters/alternating_characters.rb
def deletions(s)
  last = s[0]
  dels = 0
  for i in 1..(s.length) do
    if last == s[i]
      dels += 1
    else
      last = s[i]
    end
  end
  dels
end

var tests = gets.to_i
tests.times do
  s = gets
  puts deletions(s)
end